samples/certificate_provisioning/src/main.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(golioth_hello, LOG_LEVEL_DBG);
+
+#include <net/golioth/system_client.h>
+#include <samples/common/hardcoded_credentials.h>
+#include <samples/common/net_connect.h>
+#include <zephyr/net/coap.h>
+#include <zephyr/device.h>
+#include <zephyr/fs/fs.h>
+#include <zephyr/fs/littlefs.h>
+
+#define CLIENT_CERTIFICATE_PATH "/lfs1/credentials/client_cert.der"
+#define PRIVATE_KEY_PATH        "/lfs1/credentials/private_key.der"
+
+#define STORAGE_PARTITION_LABEL	storage_partition
+#define STORAGE_PARTITION_ID	FIXED_PARTITION_ID(STORAGE_PARTITION_LABEL)
+
+FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(cstorage);
+static struct fs_mount_t littlefs_mnt = {
+	.type = FS_LITTLEFS,
+	.fs_data = &cstorage,
+	.storage_dev = (void *)STORAGE_PARTITION_ID,
+	.mnt_point = "/lfs1"
+};
+
+static struct golioth_client *client = GOLIOTH_SYSTEM_CLIENT_GET();
+
+static K_SEM_DEFINE(connected, 0, 1);
+
+static void golioth_on_connect(struct golioth_client *client)
+{
+	k_sem_give(&connected);
+}
+
+static int load_credential_from_fs(const char *path, enum tls_credential_type type)
+{
+	struct fs_file_t file;
+	struct fs_dirent dirent;
+
+	fs_file_t_init(&file);
+
+	int err = fs_stat(path, &dirent);
+
+	if (err < 0) {
+		LOG_WRN("Could not stat %s, err: %d", path, err);
+		goto finish;
+	}
+	if (dirent.type != FS_DIR_ENTRY_FILE) {
+		LOG_ERR("%s is not a file", path);
+		err = -EISDIR;
+		goto finish;
+	}
+	if (dirent.size == 0) {
+		LOG_ERR("%s is an empty file", path);
+		err = -EINVAL;
+		goto finish;
+	}
+
+
+	err = fs_open(&file, path, FS_O_READ);
+
+	if (err < 0) {
+		LOG_ERR("Could not open %s", path);
+		goto finish;
+	}
+
+	/* NOTE: cred_buf is used directly by the TLS Credentials library, and so must remain
+	 * allocated for the life of the program.
+	 */
+
+	void *cred_buf = malloc(dirent.size);
```

Comment:
Please put variable declarations at the top of function, so it is easy to see which (and how many) variables are used within function scope.